R/defineModel.R

```diff
@@ -1,18 +1,118 @@
+defineModel <- function(dat, items, id, splittedModels = NULL,
+                        irtmodel = c("1PL", "2PL", "PCM", "PCM2", "RSM", "GPCM", "2PL.groups", "GPCM.design", "3PL"),
+                        qMatrix=NULL, DIF.var=NULL, HG.var=NULL, group.var=NULL,
+                        weight.var=NULL, anchor = NULL, domainCol=NULL, itemCol=NULL,
+                        valueCol=NULL,check.for.linking = TRUE, minNperItem = 50,
+                        removeMinNperItem = FALSE, boundary = 6, remove.boundary = FALSE,
+                        remove.no.answers = TRUE, remove.no.answersHG = TRUE,
+                        remove.missing.items = TRUE, remove.constant.items = TRUE,
+                        remove.failures = FALSE, remove.vars.DIF.missing = TRUE,
+                        remove.vars.DIF.constant = TRUE, verbose=TRUE, software = c("conquest","tam"),
+                        dir = NULL, analysis.name, schooltype.var = NULL, model.statement = "item",
+                        compute.fit = TRUE, pvMethod = c("regular", "bayesian"),
+                        fitTamMmlForBayesian = TRUE, n.plausible=5, seed = NULL,
+                        conquest.folder=NULL, constraints=c("cases","none","items"),
+                        std.err=c("quick","full","none"), distribution=c("normal","discrete"),
+                        method=c("gauss", "quadrature", "montecarlo", "quasiMontecarlo"),
+                        n.iterations=2000, nodes=NULL, p.nodes=2000, f.nodes=2000,
+                        converge=0.001, deviancechange=0.0001, equivalence.table=c("wle","mle","NULL"),
+                        use.letters=FALSE, allowAllScoresEverywhere = TRUE, guessMat = NULL,
+                        est.slopegroups = NULL, fixSlopeMat = NULL, slopeMatDomainCol=NULL,
+                        slopeMatItemCol=NULL, slopeMatValueCol=NULL, progress = FALSE,
+                        Msteps = NULL, increment.factor=1 , fac.oldxsi=0,
+                        export = list(logfile = TRUE, systemfile = FALSE, history = TRUE,
+                                      covariance = TRUE, reg_coefficients = TRUE, designmatrix = FALSE) ){
+
+### checking/asserting the arguments -------------------------------------------
+  # dat
+  ismc <- attr(dat, "multicore")
+  dat  <- eatTools::makeDataFrame(dat, name = "dat")
+
+  # list: splitted Models
+  checkmate::assert_list(splittedModels, null.ok = TRUE)
+  # subset: irtmodel
+  checkmate::assert_subset(irtmodel, choices = c("1PL", "2PL", "PCM", "PCM2","RSM",
+                                                 "GPCM", "2PL.groups", "GPCM.design", "3PL"))
+
+  # data frame: qMatrix
+  checkQmatrixConsistency(qMatrix)
+
+  #' assert vector:
+  #' DIF.var, HG.var, group.var, weight.var, schooltype.var
+  checkmate::assert_vector(DIF.var, len = 1)
```

Comment:
So now, because of this line, the checks fail. DIF.var must be a vector, not NULL. The Default is NULL, but as it is no optional argument, I didn't write null.ok = TRUE. I think this (and probably all non optional arguments with a NULL default) arguments would need actual default input, or I'll just either remove the assertion completety or add null.ok.
What do you think?

Reply:
Mhm, but DIFvar is still an optional argument. The user is free to omit it. I would suggest removing the assertion.